File: src/main/java/de/thm/arsnova/web/exceptions/UnauthorizedException.java
package de.thm.arsnova.web.exceptions;

/**
 * Unauthorized means status code 401.
 */
public class UnauthorizedException extends RuntimeException {
	private static final long serialVersionUID = 1L;

	public UnauthorizedException() {
		super();
	}

	public UnauthorizedException(String message) {
		super(message);
	}

	public UnauthorizedException(Throwable e) {
		super(e);
	}

	public UnauthorizedException(String message, Throwable e) {
		super(message, e);
	}
}

File: gateway/src/main/kotlin/de/thm/arsnova/service/httpgateway/exception/UnauthorizedException.kt
package de.thm.arsnova.service.httpgateway.exception

import org.springframework.http.HttpStatus
import org.springframework.web.server.ResponseStatusException

class UnauthorizedException : ResponseStatusException {
  constructor() : super(HttpStatus.UNAUTHORIZED)
  constructor(reason: String?) : super(HttpStatus.UNAUTHORIZED, reason)
  constructor(reason: String?, cause: Throwable?) : super(HttpStatus.UNAUTHORIZED, reason, cause)
}
